src/io/flutter/run/daemon/DaemonJsonInputFilterProvider.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.run.daemon;
+
+import com.intellij.execution.filters.ConsoleInputFilterProvider;
+import com.intellij.execution.filters.InputFilter;
+import com.intellij.execution.ui.ConsoleViewContentType;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Pair;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+
+public class DaemonJsonInputFilterProvider implements ConsoleInputFilterProvider {
+
+  @NotNull
+  @Override
+  public InputFilter[] getDefaultFilters(@NotNull Project project) {
+    return new InputFilter[]{new DaemonJsonInputFilter()};
+  }
+
+  private static class DaemonJsonInputFilter implements InputFilter {
+    @Nullable
+    @Override
+    public List<Pair<String, ConsoleViewContentType>> applyFilter(String text, ConsoleViewContentType contentType) {
+      if (text.startsWith("[{") && text.endsWith("}]\n")) {
```

Comment:
@stevemessick, instead of filtering this from the console, what if we don't write the daemon / json output in the first place? The things we'd want to show there are likely just the log messages - direct output from the tool and the user's app (the `message.log` below).